src/sentry/utils/sdk.py

```diff
@@ -445,11 +454,41 @@ def _kwargs_into_scope(self, scope, extra=None, tags=None, fingerprint=None, req
             scope.fingerprint = fingerprint
 
 
+def check_tag(tag_key: str, expected_value: str) -> bool:
+    """Detect a tag already set and being different than what we expect.
+
+    This function checks if a tag has been already been set and if it differs
+    from what we want to set it to.
+    """
+    with configure_scope() as scope:
+        if scope._tags and tag_key in scope._tags and scope._tags[tag_key] != expected_value:
+            extra = {
+                f"previous_{tag_key}": scope._tags[tag_key],
+                f"new_{tag_key}": expected_value,
+            }
+            logger.warning(f"Tag already set and different ({tag_key}).", extra=extra)
+            return True
+
+
+# We have some events being tagged with organization.slug even when we don't have such info
+def log_if_tags_already_set(scope: Scope, org_id: int, org_slug: str) -> None:
+    """If the tags are already set and differ, report them as a Sentry error."""
+    try:
+        if check_tag("organization", org_id) or check_tag("organization.slug", org_slug):
+            raise Exception(
+                "This is an intended error for investigating what is the root of this problem."
+            )
+    except Exception as e:
+        logger.exception(e)
```

Comment:
The reason I create Sentry errors is because I get access to the stacktrace and logs.
It would be great if I could make expected errors not to show up as "red" like real errors.